llpc/util/llpcUtil.cpp

```diff
@@ -122,4 +125,54 @@ unsigned shaderStageToMask(ShaderStage stage) {
   return (1 << stage);
 }
 
+// =====================================================================================================================
+// Returns true if shaderInfo has the information required to compile an unlinked shader of the given type.
+//
+// @param type : The unlinked shader type.
+// @param shaderInfo : The shader information to check.
+bool hasDataForUnlinkedShaderType(Vkgc::UnlinkedShaderStage type, ArrayRef<const PipelineShaderInfo *> shaderInfo) {
+  switch (type) {
+  case UnlinkedStageVertexProcess:
+    return ShaderStageVertex < shaderInfo.size() && shaderInfo[ShaderStageVertex] &&
```

Comment:
Can you hoist this logic into a lambda or a helper function? The current implementation seems prone to copy-paste-edit errors where someone forgets to update one part of the condition.